syft/workers/plan.py

```diff
@@ -73,12 +77,66 @@ def method(self: object) -> Plan:
     return method
 
 
-class PlanPointer:
-    def __init__(self, id, location, id_at_location, owner):
-        self.id = id
+class PlanPointer(AbstractTensor):
```

Comment:
Why is plan pointer a tensor?

Reply:
This was mainly a workaround, to be able to search tensors and plans without distinction in the code. In the [search function at base.py](https://github.com/OpenMined/PySyft/blob/8f9263fa884c2ba7f2442f2108069b592b348e1a/syft/workers/base.py#L753) they create the pointer and wrap it.

Another possibile, and potentially more intuitive, approach would be to treat them differently in the search function and make PlanPointer just a wrapper that points to the Plan.

Reply:
One of the things about plans that always stood out for me was that they were callables so plan(tensor) worked. Is that still the case with plan pointers?

Reply:
Not really, a plan pointer is not callable.

Reply:
In that case what is the intended use of plan pointer, are they what is returned when you call search for a plan, then you need to call get to use it? If that is the case, I think we should just return the plan

Reply:
> In that case what is the intended use of plan pointer, are they what is returned when you call search for a plan, then you need to call get to use it

Yes.

>  If that is the case, I think we should just return the plan

sgtm.

Reply:
So there is a bit of a confusion about Plan Pointer, the only reason why I had kept them is that they were convenient for serialisation, They don't behave as PointerTensor

Reply:
Yep, but I believe we still need some kind of wrapper around the plan to something similar to [what is being done to tensors](https://github.com/OpenMined/PySyft/blob/8f9263fa884c2ba7f2442f2108069b592b348e1a/syft/workers/base.py#L753), right? Or we should update the plan's location/owner on the plan itself?

Reply:
That's not strictly needed, we could "update the plan's location/owner on the plan itself?" but overall yes I agree we should find a way so that the Plan sending / and getting is closer to the way tensors are handled, even if for Plan the sending is lazy and doesn't happen for real until you need a command to be executed remotely

Reply:
For now, I removed the PlanPointer, and the plan itself is returned.

Reply:
With that in mind we might just want to get rid of plan pointer entirely (is it used for anything?)

Reply:
I think it's being used for serialization only. I'll try to get rid of it entirely.

Reply:
Actually, since this can make this PR a little to big and removing this pointer must be done carefully, I'll just remove my changes, and keep it as a simple wrapper.